lib/kafka/message_buffer.rb

```diff
@@ -48,6 +48,40 @@ def each
       end
     end
 
+    def clear_largest_message
+      target = nil
+      @buffer.each do |topic, messages_for_topic|
+        messages_for_topic.each do |partition, messages_for_partition|
+          messages_for_partition.each do |message|
+            if target.nil? || target[:message].bytesize.to_i < message.bytesize.to_i
+              target = {
+                topic: topic,
+                partition: partition,
+                message: message,
+              }
+            end
+          end
+        end
+      end
+
+      clear_message(target) unless target.nil?
```

Comment:
```suggestion
      clear_message(**target) unless target.nil?
```

This will cause a warning in Ruby 2.7 otherwise, I believe.